en/companion_computer/pixhawk_rpi.md

```diff
@@ -0,0 +1,323 @@
+In the next section we'll set up the both Pixhawk and RPi to use XRCE-DDS and ROS2 instead of MAVLink.
+
+## ROS 2 and XRCE-DDS
+
+The [ROS 2 Guide](../ros/ros2_comm.html) and [XRCE-DDS](../middleware/xrce_dds.md) pages cover the options for setting up the XRCE-DDS and ROS, focussing on ROS 2 "Foxy".
+This tutorial uses ROS 2 "Humble" and covers the specific setup for working with RPi.
+It is worth reading both!
+
+### Pixhawk/PX4 Setup
+
+Next we set up ROS 2 instead of MAVLink on `TELEM2`.
+We do this by changing parameters in QGroundControl, which can be connected via USB, or using a telemetry radio connected to `TELEM1`.
+
+The configuration steps are:
+
+1. Connect the Pixhawk with the laptop using a USB cable and open QGroundControl (if not currently connected).
+1. [Check/change the following parameters](../advanced_config/parameters.md) in QGroundControl:
+
+   ```
+   MAV_1_CONFIG = 0 (Disabled)
+   XRCE_DDS_CFG = 102 (TELEM2)
+   SER_TEL2_BAUD = 921600
+   ```
+
+   [MAV_1_CONFIG=0](../advanced_config/parameter_reference.md#MAV_1_CONFIG) and [XRCE_DDS_CFG=102](../advanced_config/parameter_reference.md#MAV_1_CONFIG) disable MAVLink on TELEM2 and enable the XRCE-DDS client on TELEM2, respectively.
+   The `SER_TEL2_BAUD` rate sets the comms link data rate.
+   
+   Note that you could similarly configure a connection to `TELEM1`.
+1. Check that the [microdds_client](../modules/modules_system.md#microdds-client) module is now running.
+   YOu can do this by running the following command in the QGroundControl [MAVLink Console](https://docs.qgroundcontrol.com/master/en/analyze_view/mavlink_console.html): 
+
+   ```
+   microdds_client status
+   ```
+   
+:::note
+If the client module is not running you can start it manually in the MAVLink console: 
+
+```
+microdds_client start -t serial -d /dev/ttyS3 -b 921600
+```
+
+Note that `/dev/ttyS3` is the PX4 port for `TELEM2` on the [Holybro Pixhawk 6c](../flight_controller/pixhawk6c.html#serial-port-mapping).
+For other flight controllers check the serial port mapping section in their overview page.
+:::
+
+### ROS Setup on RPi
+
+The steps to setup ROS 2 and the XRCE-DDS Agent on the RPi are:
+
+1. Install ROS 2 Humble by following the [official tutorial](https://docs.ros.org/en/humble/Installation/Ubuntu-Install-Debians.html).
+2. Install the git using the RPi terminal:
+
+   ```bash
+   sudo apt install git
+   ```
+3. Install the XRCE_DDS client:
+
+   ```bash
+   git clone https://github.com/eProsima/Micro-XRCE-DDS-Agent.git
+   cd Micro-XRCE-DDS-Agent
+   mkdir build
+   cd build
+   cmake ..
+   make
+   sudo make install
+   sudo ldconfig /usr/local/lib/
+   ```
+   
+   See [XRCE-DDS > XRCE-DDS Agent Installation](../middleware/xrce_dds.md#xrce-dds-agent-installation) for alternative ways of installing the agent.
+4. Start the agent in the RPi terminal:
+
+   ```bash
+   sudo MicroXRCEAgent serial --dev /dev/serial0 -b 921600
```

Comment:
FYI @hamishwillee , `sudo` is used here and also in the main XRCE-DDS page. However, I think that it is not necessary as long as the user running `MicroXRCEAgent` has the right permissions to access the serial device.

Reply:
@beniaminopozzan I don't plan to change this. Not because I think you're wrong, but because:
- nothing ever works for me on Ubuntu without sudo, 
- while Linux experts complain, there doesn't seem to be any reason not to.
- If someone understand Linux they will manage to install without sudo. But if you don't particularly, then you'll do this anyway.

Upshot, if someone wants to retest all this and verify it all works without sudo then great.